subprojects/docs/src/docs/userguide/upgrade_version_4.adoc

```diff
@@ -0,0 +1,483 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[upgrade_version_4]]
+= Upgrading Gradle Version 4.x to 5.0
+
+This chapter provides the information you need to migrate your older Gradle 4.x builds to the latest version. In most cases, you will need to apply the changes from all versions that come after the one you're upgrading from. For example, if you're upgrading from Gradle 4.3, you will also need to apply the changes from 4.4, 4.5, etc.
+
+A few pointers about upgrading:
+
+ * Run `gradle help` with the version of Gradle you're upgrading to in order to see any deprecations that apply to your build.
+ * Make sure you update the versions of plugins you use in your build when upgrading the Gradle version. Some plugins will break with newer versions of Gradle.
+ * In most cases, deprecated features, properties and methods will be removed in Gradle 5.0, resulting in build errors if you try to use them.
+
+The most significant changes include the following:
+
+ * A recommendation to <<#rel4.7:switch_to_publishing_plugins, switch to the Maven Publish and Ivy Publish plugins>> if you're not already using them. They now support digital signatures with the Signing Plugin.
+ * A change that means you should <<#rel4.7:configure_internal_tasks,configure existing `wrapper` and `init` tasks>> rather than defining your own.
+ * The <<#rel4.7:pom_wildcard_exclusions,honoring of implicit wildcards in Maven POM exclusions>>, which may result in dependencies being excluded that weren't before.
+ * A <<#upgrade_4.5,recommendation to enable the new and improved POM support>> in your builds.
+ * A <<#rel4.5:annotation_processor_configuration,change to the way you add Java annotation processors to a project>>.
+ * The <<custom_tasks.adoc#worker_api,Worker API>> for enabling units of work to execute in parallel.
```

Comment:
Note that the '.adoc' in the link is unnecessary. Works for both multi-page and single-page without it.